software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java

```diff
@@ -56,12 +60,17 @@ public void start(Collection<? extends Location> locations) {
     
     public static class MemberTrackingPolicy extends AbstractMembershipTrackingPolicy {
         @Override protected void onEntityEvent(EventType type, Entity member) {
+            defaultHighlightAction(type, entity);
             ((MongoDBRouterClusterImpl)super.entity).setAnyRouter();
         }
         @Override protected void onEntityRemoved(Entity member) {
+            // TODO shouldn't be invoked - remove
+            log.warn("Removal handler should be hidden by event handler", new Throwable("Trace for unexpected mongo node handler"));
```

Comment:
Not sure I understand what this is for: is this an error per se? Looks like a removal is handled correctly so why the warn message? Can you give a bit more details ?

Reply:
more a note to developer (it caught me out) - due to the `onEntityEvent` handler being overridden this method is not used

Reply:
That make sense now, as `onEntityEvent` does not call `super`